test/utils.h

```diff
@@ -382,6 +382,34 @@ class ContextCudnnTF32Disabled {
   bool flag_;
 };
 
+inline bool maybeClearAllocator(int64_t max_bytes = ((int64_t)1 << 32)) {
+  // check used memory and empty allocator cache if above a set threshold
+  auto allocator = c10::cuda::CUDACachingAllocator::get();
+  if (allocator->initialized()) {
+    int device = 0;
+#define TORCH_VERSION_GREATER(major, minor, patch)                    \
+  TORCH_VERSION_MAJOR > major ||                                      \
+      (TORCH_VERSION_MAJOR == major && TORCH_VERSION_MINOR > minor || \
+       (TORCH_VERSION_MINOR == minor && TORCH_VERSION_PATCH > patch))
```

Comment:
Perhaps this belongs in `csrc/utils.h`. I didn't see any, but thought maybe there is already such a macro in PyTorch or NVFuser? @jjsjann123